src/core/image_processor.py

```
@@ -211,353 +212,191 @@ def detect_neurons(
+        suffix = path.suffix.lower()
         
-        # Convert back to original dtype, preserving the original value range
-        if output_dtype == np.uint8:
-            aligned = np.clip(aligned_float, 0, 255).astype(np.uint8)
-        elif output_dtype == np.uint16:
-            aligned = np.clip(aligned_float, 0, 65535).astype(np.uint16)
+        # Use tifffile for TIFF files (supports 16-bit, multi-page, etc.)
+        if suffix in ['.tif', '.tiff']:
+            try:
+                img = tifffile.imread(str(path))
+            except Exception as e:
```

Ruff (0.14.3)

Do not catch blind exception: `Exception`

(BLE001)